### PR TITLE
NF: read segmentation data from fda files

### LINKED ISSUE #94: Parse segmentation from fda files
Hi and thanks for the tool.

I am trying to parse segmentation data from fda files to extend [eyepy](https://github.com/MedVisBonn/eyepy/discussions/11).

I believe the segmentation is stored in @CONTOUR_INFO chunks (one per layer).

I've noticed that the [`read_any_info_and_make_dict`](https://github.com/marksgraham/OCT-Converter/blob/155df23327065214013036055a4d91d791998f8b/oct_converter/readers/fda.py#LL158C11-L158C55) cannot be used to retrieve the segmentation of all layers as they share the same name and key in the dictionary.

What do you think about adding a `read_segmentation()` method to `FDA` class ? I can do it myself if it sounds good.

### PR DESCRIPTION
**Pull request by drombas**
Closes #94 

Changes
- Add a `read_segmentation` method to FDA class
- Update fda example

Comments
- I had to change the chunk listing to store the position of all @CONTOURINFO chunks as a list
- I have noticed that OCTVolumeWithMetaData class supports contour information. Maybe for a next PR?

Happy to make any changes if needed.

### CODE REVIEW

**Comment by marksgraham** [comment]
Hi @drombas 

Thanks for this! I'm snowed under with a conference deadline until Friday, so I'll take a proper look at it then. 

You're right, there is a contours part of OCTVolume, storing it there would make viewing them easily compatible with the `peek` method as we've done for e2e [here](https://github.com/marksgraham/OCT-Converter/blob/main/examples/demo_e2e_extraction.py), which would be nice! But we can save for another PR.

File: oct_converter/readers/fda.py [hunk 1]
```diff
@@ -155,6 +166,52 @@ def read_fundus_image_gray_scale(self):
...
+        """
+        Reads layer segmentation data.
+
+        Segmentation values are returned in a dictionary with a key for every
+        layer boundary and are measured in pixels from B-scan bottom.
+
+        Returns
+        -------
+            seg_dict: dictionary with segmentation data.
+        """
```
Comment (marksgraham) [inline oct_converter/readers/fda.py:179]: Please could you change to the following to make compatible with the other docstrings:

```
        Returns
            dict: dictionary with segmentation data.
```

**Review by marksgraham — CHANGES_REQUESTED** [review]
Hey, looks great! 

It looks like its easy to get this compatible with the `OCTVolume` contours field. Would you be able to add this to the `read_oct_volume` method?

```
        # read segmentation contours and store in OCTVolume, if possible
        contours = self.read_segmentation()
        if contours:
            # define as distance from top of scan to make compatible with plotting in OCTVolume
            image_height = volume[:, :, i].shape[0]
            contours = {k:image_height - v for k,v in contours.items()}
        oct_volume.contours = contours
        return oct_volume
```

One that is done we can update the example with `oct_volume.peek(show_contours=True)` and get out something like this:

image

This requires the layers to be defined from the top of the b-scan, rather than bottom, to stay consistent with how it is done in the E2E class. I propose we leave the `read_segmentation` function as is and do the transformation before it gets stored in the OCTVolume, allowing you to get contours defined from the bottom for compatibility with `eyepy`

**Comment by drombas** [comment]
Hi @marksgraham. Thanks a lot for your feedback. 

I've made the changes. I took the liberty of simplifying the code a bit but feel free to request any changes.

**Review by marksgraham — APPROVED** [review]
Looks great :)

**Pull request merged**